### ISSUE TITLE
How to communicate with the backend in Authoring Theia Extensions.

### ISSUE THREAD

**Issue by wonjooncho** [theia-ide/theia#5721]
### Description
I am creating an extension with reference to [Authoring Theia Extensions](https://www.theia-ide.org/doc/Authoring_Extensions.html), [Json-Rpc](https://www.theia-ide.org/doc/Json_Rpc.html) and [Complete example in Json-Rpc Docs](https://github.com/theia-ide/theia/commit/99d191f19bd2a3e93098470ca1bb7b320ab344a1).
I added a menu and when I select that menu I want to call the backend using json-rpc, but it does not get called.
The server object variable I was calling was empty.
In my case, I used WorkspaceServer in my backend server code.

And I do not know how to add 'services / *' to the webpack.config.js file.
After I edit the file, I build it as 'yarn' and all of my changes disappear.
![Uploading image.png…]()

Is there any additional documentation or source code example that I can refer to to solve this problem?

**Comment by kittaakos** [comment]
Can you share your code?

**Comment by wonjooncho** [comment]
@kittaakos Share the source code [here](https://github.com/wonjooncho/theia-backend).

**Comment by kittaakos** [comment]
> In my case, I used WorkspaceServer in my backend server code.

Why the `WorkspaceServer`? What do you want to do? I thought you want to create a service that is implemented on the backend and be able to call this service from the frontend? Is this what you want to achieve?

**Comment by wonjooncho** [comment]
@kittaakos I use WorkspaceServer because I refer to other source codes like that @theia/core, and there is no other reason.
I just want to test server side code.

What I want to do is get the server side environment variables and pass them to the client(frontend).
And I want to do REST-API processing on the server site and pass the result to the client(frontend).

**Comment by kittaakos** [comment]
> I use WorkspaceServer because I refer to other source codes like that @theia/core, and there is no other reason.

OK. So if I got this correctly, it is not a must to use the `WorkspaceService`.

> I just want to test server side code.

👍
I have created a PR with the changes of a simple RPC service example: https://github.com/wonjooncho/theia-backend/pull/1

![screencast 2019-07-16 11-07-17](https://user-images.githubusercontent.com/1405703/61282679-10db2b80-a7bc-11e9-9382-5b7fb557da9d.gif)

> And I want to do REST-API processing on the server site and pass the result to the client(frontend).

You will need a cutom `BackendApplicationContribution` for this. In the `configure(app: express.Application)` method you register your backend REST endpint. Check for an `express` example. For instance, here: https://www.codementor.io/wapjude/creating-a-simple-rest-api-with-expressjs-in-5min-bbtmk51mq.

I hope this helps.
